sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -5257,11 +5257,11 @@ pi_result cuda_piextUSMGetMemAllocInfo(pi_context context, const void *ptr,
 // pi_level_zero.cpp for reference) Currently this is just a NOOP.
 pi_result cuda_piTearDown(void *) { return PI_SUCCESS; }
 
-const char SupportedVersion[] = _PI_H_VERSION_STRING;
+const char SupportedVersion[] = _PI_CUDA_PLUGIN_VERSION_STRING;
 
 pi_result piPluginInit(pi_plugin *PluginInit) {
-  int CompareVersions = strcmp(PluginInit->PiVersion, SupportedVersion);
-  if (CompareVersions < 0) {
+  static int PiVersionLen = strlen(PluginInit->PiVersion);
+  if (strncmp(PluginInit->PiVersion, SupportedVersion, PiVersionLen) < 0) {
```

Comment:
`strncmp` only compares the first `n` characters of the string, so in this case is only checking the major and minor versions (whose string uses `PiVersionLen` characters). I will change this so that it only checks the major version